cardano-node/src/Cardano/Node/Run.hs

```diff
@@ -105,6 +106,23 @@ runNode
   :: PartialNodeConfiguration
   -> IO ()
 runNode cmdPc = do
+    -- Workaround to ensure that the main thread throws an async exception on
+    -- receiving a SIGTERM signal.
+#ifdef UNIX
+    -- Similar implementation to the RTS's handling of SIGINT (see GHC's
+    -- TopHandler.hs)
+    runThreadIdWk <- mkWeakThreadId =<< myThreadId
+    void $ Signals.installHandler
+      Signals.sigTERM
+      (Signals.CatchOnce $ do
+        runThreadIdMay <- deRefWeak runThreadIdWk
+        case runThreadIdMay of
+          Nothing -> return ()
+          Just runThreadId -> killThread runThreadId
+      )
+      Nothing
+#endif
```

Comment:
This function is fairly large already.  Can this be moved to a separate function with a meaningful name and called from here?

Reply:
Will do in https://github.com/input-output-hk/cardano-node/pull/3641